File: kinotic-management-api/src/main/java/org/kinotic/management/api/model/MicroserviceArtifact.java
package org.kinotic.management.api.model;

/**
 * One microservice artifact of a project: a package directly under
 * {@code packages/microservices} of the project's checkout.
 *
 * @param name  the artifact's identity, the unscoped part of the {@code name} in the package's
 *              {@code package.json}: {@code @acme/orders} is {@code orders}. Always a single
 *              zone label
 * @param dir   the package directory, relative to the checkout root
 * @param entry the module the runtime starts, relative to {@link #dir}: the
 *              {@code package.json} {@code main}, or {@code src/main.ts} when it declares none
 */
public record MicroserviceArtifact(String name, String dir, String entry) {}

File: kinotic-js/workspace/packages/management-api/src/api/model/MicroserviceArtifact.ts
/**
 * One microservice artifact of a project: a package directly under packages/microservices
 * of the project's checkout.
 */
export interface MicroserviceArtifact {
    /**
     * The artifact's identity, the unscoped part of the name in the package's package.json:
     * "@acme/orders" is "orders". Always a single zone label.
     */
    name: string
    /**
     * The package directory, relative to the checkout root.
     */
    dir: string
    /**
     * The module the runtime starts, relative to dir: the package.json main, or src/main.ts
     * when it declares none.
     */
    entry: string
}
